ledger-tool/src/main.rs

```diff
@@ -1817,11 +1817,7 @@ fn main() {
                         working_bank.print_accounts_stats();
                     }
                     if print_bank_hash {
-                        println!(
-                            "Bank hash for slot {}: {}",
-                            working_bank.slot(),
-                            working_bank.hash()
-                        );
+                        println!("{}", working_bank.hash());
```

Comment:
Oops, is there any possibility that someone does:
agave-ledger-tool verify --print-bank-hash --print_something_else?

If that's the case we still need the other information.

Reply:
As far as I know, the `bank-hash` and `shred-version` command are only used by the `remote-node.sh` script. As someone who uses ledger-tool quite a bit, I rarely (if ever) use these commands.

But yes, you're right that if someone printed out a bunch of stuff (bank hash, accounts stats, shred-version, etc) it could be confusing as to what is what. In general, I am less enthusiastic about deprecating `bank-hash` and `shred-version` commands now.

@CriesofCarrots - knowing what we know now, are you still onboard with deprecating `bank-hash` and `shred-version` commands ?

Reply:
> knowing what we know now, are you still onboard with deprecating bank-hash and shred-version commands ?

It still seems reasonable to me. How much trouble would it be to support `--output json` to make it easier for `remote-node.sh` to find the data it needs (and leave the descriptive labels for the human-readable version)?

Reply:
Yeah, I don't know how often human would need this, multinode-demo is used frequently enough so it would help to have a human API.

Maybe we can just output one json which always output bank-hash and shred-version. Would it hurt performance if we read both of them? I suppose this json is small enough it should be human readable.

Reply:
>  How much trouble would it be to support --output json

Now much trouble at all. I think parsing json in bash could be messy, so am I correct in assuming that we'd be ok with using `jq` in this script ?

> Would it hurt performance if we read both of them?

There is a second call in this script to get the `shred-version`. On a MNB ledger, having separate calls costs you minutes. On a brand new ledger for bootstrap, it costs less than a second

Reply:
Might be okay to use jq, but it would be best not having to parse anything at all. Machine really needs --output raw.

Hmm, if it costs minutes more, then let's still separate the reading of two.

Reply:
I think Steve is saying it will be much more performant to get both values in the same call.

> am I correct in assuming that we'd be ok with using jq in this script ?

We'll need to ensure test clusters have `jq` installed, but yeah, I think that's the approach.

Reply:
Requiring test clusters to have `jq` installed sounds like a big hassle though? IMHO it's not worth the performance gains by getting both values in the same call.

Reply:
> Hmm, if it costs minutes more, then let's still separate the reading of two.

To confirm, the minutes more for calling `ledger-tool` twice would be for a MNB ledger. Namely, unpacking a snapshot into a `Bank` twice

> I think Steve is saying it will be much more performant to get both values in the same call.

For MNB, definitely. For a brand new test cluster, we're reading `genesis.bin` and replaying slot 0 (all ticks). So, one vs. two `ledger-tool` invocations is neglible.

> Requiring test clusters to have jq installed sounds like a big hassle though?

This isn't immediately clear to me. I'll need to skim the scripts again. Given that you found this issue, is it fair to assume that you're currently using these scripts ? If so, are you deploying on bare-metal, GCE, ... ?

> IMHO it's not worth the performance gains by getting both values in the same call.

For now, I think we can probably ignore the issue of returning both shred version and bank hash in one call, and just focus on the "how can we easily parse this in a script" issue.

Reply:
> This isn't immediately clear to me. I'll need to skim the scripts again. Given that you found this issue, is it fair to assume that you're currently using these scripts ? If so, are you deploying on bare-metal, GCE, ... ?
> 
I'm using it on GCE, I can confirm `jq` isn't installed by default. Not a big hassle to install it, but it's one more dependency.

Reply:
> I'm using it on GCE, I can confirm jq isn't installed by default

Good to know, thanks for checking.

> Not a big hassle to install it, but it's one more dependency.

True, I think `jq` is a fairly ubiquitous tool. Looks like it'd be pretty easy to install it to, just make another script like one of these (for all deployment types):
https://github.com/anza-xyz/agave/blob/dbea20b12aa5124400fad145343f34dd92e1d7d5/net/gce.sh#L804-L813

Reply:
> > I'm using it on GCE, I can confirm jq isn't installed by default
> 
> Good to know, thanks for checking.
> 
> > Not a big hassle to install it, but it's one more dependency.
> 
> True, I think `jq` is a fairly ubiquitous tool. Looks like it'd be pretty easy to install it to, just make another script like one of these (for all deployment types):
> 
> https://github.com/anza-xyz/agave/blob/dbea20b12aa5124400fad145343f34dd92e1d7d5/net/gce.sh#L804-L810

I'm fine with that. How about adding install-jq.sh to this PR so we know why the dependency is introduced?

Reply:
Apologies about the multiple notifications for pushing commits - should have copied the patch files over instead of pushing the branch to test.

In any case, think what I have in this PR now does what we agreed on